pkg/server/api/entry/v1/service_test.go

```diff
@@ -799,6 +799,448 @@ func setupServiceTest(t *testing.T, ds datastore.DataStore) *serviceTest {
+						logrus.ErrorKey: "invalid spiffe ID: spiffeid: trust domain is empty",
+						"spiffe_id":     "path:\"/invalid\" ",
+					},
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.InvalidArgument),
+						Message: "failed to convert entry: invalid spiffe ID: spiffeid: trust domain is empty"},
+				},
+			},
+		},
+		{
+			name:           "Fail Invalid Parent Id",
+			initialEntries: []*types.Entry{initialEntry},
+			inputMask: &types.EntryMask{
+				ParentId: true,
+			},
+			updateEntries: []*types.Entry{
+				{
+					ParentId: &types.SPIFFEID{TrustDomain: "", Path: "/invalid"},
+				},
+			},
+			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "failed to convert entry",
+					Data: logrus.Fields{
+						logrus.ErrorKey: "invalid parent ID: spiffeid: trust domain is empty",
+					},
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.InvalidArgument),
+						Message: "failed to convert entry: invalid parent ID: spiffeid: trust domain is empty"},
+				},
+			},
+		},
+		{
+			name:           "Fail Datastore Error",
+			initialEntries: []*types.Entry{initialEntry},
+			inputMask: &types.EntryMask{
+				ParentId: true,
+			},
+			updateEntries: []*types.Entry{
+				{
+					ParentId: &types.SPIFFEID{TrustDomain: "", Path: "/invalid"},
+				},
+			},
+			dsError: errors.New("some error"),
+			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "failed to fetch entry",
+					Data: logrus.Fields{
+						logrus.ErrorKey: "some error",
+					},
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.Internal), Message: "failed to fetch entry"},
+				},
+			},
+		},
+		{
+			name:           "Check Output Masks Work",
+			initialEntries: []*types.Entry{initialEntry},
+			inputMask:      &types.EntryMask{
+				// With this empty, the update operation should be a no-op
+			},
+			outputMask: &types.EntryMask{
+				SpiffeId: true,
+			},
+			updateEntries: []*types.Entry{
+				{
+					Ttl: 500000,
+				},
+			},
+			expectResults: []*entrypb.BatchUpdateEntryResponse_Result{
+				{
+					Status: &types.Status{Code: int32(codes.OK), Message: "ok"},
+					Entry: &types.Entry{
+						SpiffeId: &types.SPIFFEID{TrustDomain: "example.org", Path: "/workload"},
+					},
+				},
+			},
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			test.logHook.Reset()
+
+			// First create the initial entries
+			createResp, err := test.client.BatchCreateEntry(ctx, &entrypb.BatchCreateEntryRequest{
+				Entries:    tt.initialEntries,
+				OutputMask: tt.outputMask,
+			})
+			require.NoError(t, err)
+			require.Equal(t, len(createResp.Results), len(tt.updateEntries))
+			// Be sure to clean up the entries created
+			defer func() {
```

Comment:
maybe instead of trying to clean all, you can just create setup new test on each iteration? and create what you need? that will siimplify and be sure that we are always running what we expect

Reply:
Good idea! Done